### DISCUSSION TITLE
DBeaver with AWS Redshift

### DISCUSSION THREAD

**Discussion by alexvto** [dbeaver/dbeaver#34904] [DBeaver Development]
DBeaver does not report currently the INOUT parameters of Redshift procedures (nor is the DDL correct in this respect).

This is the SP DDL as created (and reported by the AWS RS query editor v2):

CREATE OR REPLACE PROCEDURE ohrs.av_proc(p1 integer, INOUT p2 integer, INOUT p3 character varying(256))
 LANGUAGE plpgsql
AS $$
begin
    p2 := p1;
    p3 := to_char(p1);
end;
$$

In DBeaver, both the parameters tab and the Source (DDL) tab report all parameters as IN only:

-- DROP PROCEDURE ohrs.av_proc(int4, int4, varchar);

CREATE OR REPLACE PROCEDURE ohrs.av_proc(p1 int4, p2 int4, p3 varchar)
	LANGUAGE plpgsql
AS $$
	
begin
    p2 := p1;
    p3 := to_char(p1);
end;

$$
;

I am attaching a few screenshots as proof:

![Actual_RS_SP_DDL](https://github.com/user-attachments/assets/5a9f78ab-8c61-45ff-919a-104a67c59eec)
![DBeaver_RS_SP_DDL](https://github.com/user-attachments/assets/fd4e9e29-ad4c-4aee-bd9c-a85bede6da2c)
![DBeaver_RS_SP_function_parameters](https://github.com/user-attachments/assets/69220386-b570-4cf2-8418-2ab668f085a4)

Thank you everyone!

**Comment by E1izabeth** [comment]
Thank you for highlighting this flaw. I created the issue from this discussion #34971
